packages/builder/plugin-swc/src/plugin.ts

```diff
@@ -49,7 +49,10 @@ export const builderPluginSwc = (
         jsc: {
           transform: {
             react: {
-              refresh: env === 'development' && builderConfig.dev.hmr,
+              refresh:
```

Comment:
Maybe we can reuse this method:

![image](https://user-images.githubusercontent.com/7237365/221127360-a4d2f17e-87f6-4dc0-897e-40eccd9c5225.png)

Reply:
Yeap, Right.
Hold on, I am modifying.

Reply:
It seems many `isUsingHRM` in difference packages, e.g: `@modern-js/builder-rspack-provider`, `@modern-js/builder-webpack-provider"`.

It may be better that If we put the function into `@modern-js/builder-shared`, 
then  we let these packages, e.g `@modern-js/builder-rspack-provider`, `@modern-js/builder-plugin-swc` to import the `@modern-js/builder-shared`

Reply:
Yes, `isUsingHMR` should be moved to the `@modern-js/builder-shared` package.